
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>4.5.5. Download protein sequence database &#8212; Linux for Biologists 2021 documentation</title>
    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/alabaster.css" />
    <script data-url_root="../../" id="documentation_options" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/jquery.js"></script>
    <script src="../../_static/underscore.js"></script>
    <script src="../../_static/doctools.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="4.5.6. Search database using query sequence" href="search.html" />
    <link rel="prev" title="4.5.4. Download query sequence" href="query.html" />
   
  <link rel="stylesheet" href="../../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="download-protein-sequence-database">
<h1><span class="section-number">4.5.5. </span>Download protein sequence database<a class="headerlink" href="#download-protein-sequence-database" title="Permalink to this headline">¶</a></h1>
<p>You can follow these steps to download and prepare the
protein sequence database:</p>
<ol class="arabic simple">
<li><p>Create new directory</p></li>
<li><p>Change into it</p></li>
<li><p>Download the database archive</p></li>
<li><p>Uncompress (or extract) the database archive</p></li>
<li><p>Format the database</p></li>
</ol>
<section id="create-new-directory-mkdir">
<h2>Create new directory — <code class="docutils literal notranslate"><span class="pre">mkdir</span></code><a class="headerlink" href="#create-new-directory-mkdir" title="Permalink to this headline">¶</a></h2>
<p>In order to keep all BLAST databases in one location,
create a directory to store them using the
<a class="reference internal" href="../commands/mkdir.html"><span class="doc">mkdir</span></a> command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir ~/databases
</pre></div>
</div>
</section>
<section id="change-directory-cd">
<h2>Change directory — <code class="docutils literal notranslate"><span class="pre">cd</span></code><a class="headerlink" href="#change-directory-cd" title="Permalink to this headline">¶</a></h2>
<p>Change into the newly created directory using the
<a class="reference internal" href="../commands/cd.html"><span class="doc">cd</span></a> command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">cd</span> ~/databases
</pre></div>
</div>
</section>
<section id="download-the-database-archive-wget">
<h2>Download the database archive — <code class="docutils literal notranslate"><span class="pre">wget</span></code><a class="headerlink" href="#download-the-database-archive-wget" title="Permalink to this headline">¶</a></h2>
<p>Visit the database <a class="reference external" href="https://www.uniprot.org/downloads">downloads</a> page on the UniProt website.</p>
<p>Navigate to the <span class="guilabel">UniProtKB</span> section.</p>
<p>Right-click on the fasta download link
corresponding to <span class="guilabel">Reviewed (Swiss-Prot)</span> and then
copy it to clipboard (<a class="reference internal" href="#fig-451"><span class="std std-numref">Fig. 84</span></a>).</p>
<figure class="align-default" id="id1">
<span id="fig-451"></span><img alt="../../_images/swiss-prot-fasta.png" src="../../_images/swiss-prot-fasta.png" />
<figcaption>
<p><span class="caption-number">Fig. 84 </span><span class="caption-text">Download link for Swiss-Prot database</span><a class="headerlink" href="#id1" title="Permalink to this image">¶</a></p>
</figcaption>
</figure>
<p>To download the database, you can use the
<code class="docutils literal notranslate"><span class="pre">wget</span></code> command with the link to download as the
argument:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>wget https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz
</pre></div>
</div>
<p>When the download is complete, you will find a file
named <code class="docutils literal notranslate"><span class="pre">uniprot_sprot.fasta.gz</span></code> in the current directory.
You can use the <a class="reference internal" href="../commands/ls.html"><span class="doc">ls</span></a> command to verify
if it exists:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ls -lh
</pre></div>
</div>
<p>Output:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">total 86M</span>
<span class="go">-rw-rw-r-- 1 user user 86M Feb 10 15:00 uniprot_sprot.fasta.gz</span>
</pre></div>
</div>
<p>Since this file is in a compressed format (<code class="docutils literal notranslate"><span class="pre">.gz</span></code>),
you will need to uncompress it before proceeding.</p>
</section>
<section id="uncompress-the-database-archive-gunzip">
<h2>Uncompress the database archive — <code class="docutils literal notranslate"><span class="pre">gunzip</span></code><a class="headerlink" href="#uncompress-the-database-archive-gunzip" title="Permalink to this headline">¶</a></h2>
<p>To uncompress (or extract) the database archive file
downloaded in the previous step, you can use the
<code class="docutils literal notranslate"><span class="pre">gunzip</span></code> command.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>By default, gunzip will remove the original
compressed file after extraction.</p>
<p>If you would like to keep the original file (<code class="docutils literal notranslate"><span class="pre">.gz</span></code>),
you can include the <code class="docutils literal notranslate"><span class="pre">-k</span></code> (keep input files) option
with <code class="docutils literal notranslate"><span class="pre">gunzip</span></code>.</p>
</div>
<p>Provide the file name of the downloaded file
as the argument:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>gunzip uniprot_sprot.fasta.gz
</pre></div>
</div>
<p>When the extraction is complete, you will find the
database file in FASTA format in the same directory:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ls -lh
</pre></div>
</div>
<p>Output:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">total 267M</span>
<span class="go">-rw-rw-r-- 1 user user 267M Feb 10 15:00 uniprot_sprot.fasta</span>
</pre></div>
</div>
</section>
<section id="view-the-database">
<h2>View the database<a class="headerlink" href="#view-the-database" title="Permalink to this headline">¶</a></h2>
<p>Since this extracted database file is large,
you can use the <a class="reference internal" href="../commands/head.html"><span class="doc">head</span></a> command to
view the first few lines of the file:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>head -n <span class="m">5</span> uniprot_sprot.fasta
</pre></div>
</div>
<p>Output:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">&gt;sp|Q6GZX4|001R_FRG3G Putative transcription factor 001R OS=Frog virus 3 (isolate Goorha) OX=654924 GN=FV3-001R PE=4 SV=1</span>
<span class="go">MAFSAEDVLKEYDRRRRMEALLLSLYYPNDRKLLDYKEWSPPRVQVECPKAPVEWNNPPS</span>
<span class="go">EKGLIVGHFSGIKYKGEKAQASEVDVNKMCCWVSKFKDAMRRYQGIQTCKIPGKVLSDLD</span>
<span class="go">AKIKAYNLTVEGVEGFVRYSRVTKQHVAAFLKELRHSKQYENVNLIHYILTDKRVDIQHL</span>
<span class="go">EKDLVKDFKALVESAHRMRQGHMINVKYILYQLLKKHGHGPDGPDILTVKTGSKGVLYDD</span>
</pre></div>
</div>
<p>Alternatively, you can use the
<a class="reference internal" href="../commands/less.html"><span class="doc">less</span></a> command to view it one
page at a time:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>less uniprot_sprot.fasta
</pre></div>
</div>
<p>If you would like to count the number of sequences in the
database, you can use the <code class="docutils literal notranslate"><span class="pre">grep</span></code> command.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>grep <span class="s2">&quot;&gt;&quot;</span> -c uniprot_sprot.fasta
</pre></div>
</div>
<p>Output:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">564277</span>
</pre></div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">-c</span></code> option of <code class="docutils literal notranslate"><span class="pre">grep</span></code>, counts the number of
times the given search string (<code class="docutils literal notranslate"><span class="pre">&gt;</span></code> in this case)
occurs in the input file.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>A sequence in a FASTA format should start with
the <code class="docutils literal notranslate"><span class="pre">&gt;</span></code> character. Hence, counting the number of times
it occurs gives the number of sequences in the file.</p>
</div>
<p>You can now proceed towards formatting the database.</p>
</section>
<section id="format-the-database-makeblastdb">
<h2>Format the database — <code class="docutils literal notranslate"><span class="pre">makeblastdb</span></code><a class="headerlink" href="#format-the-database-makeblastdb" title="Permalink to this headline">¶</a></h2>
<p>The database needs to be formatted before it can be
used in a BLAST search. You can format it using the
<code class="docutils literal notranslate"><span class="pre">makeblastdb</span></code> command, which is part of the
NCBI BLAST+ package.</p>
<p>The command has multiple options. Here is an
example:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span> makeblastdb -in uniprot_sprot.fasta -parse_seqids <span class="se">\</span>
-title <span class="s2">&quot;Swiss-Prot&quot;</span> -dbtype prot -out swissprot
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The <code class="docutils literal notranslate"><span class="pre">\</span></code> character splits the long command into
multiple lines.</p>
</div>
<p id="new-db-name">Output:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go"> Building a new DB, current time: 03/24/2021 15:12:50</span>
<span class="hll"><span class="go"> New DB name:   /home/user/databases/swissprot</span>
</span><span class="go"> New DB title:  Swiss-Prot</span>
<span class="go"> Sequence type: Protein</span>
<span class="go"> Keep MBits: T</span>
<span class="go"> Maximum file size: 1000000000B</span>
<span class="go"> Adding sequences from FASTA; added 564277 sequences in 47.507 seconds.</span>
</pre></div>
</div>
<p>What the options mean:</p>
<dl class="simple">
<dt><code class="docutils literal notranslate"><span class="pre">-in</span></code></dt><dd><p>File name containing input sequences.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">-parse_seqids</span></code></dt><dd><p>Parse sequence identifiers from the input file. These
will be displayed in search results.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">-title</span></code></dt><dd><p>A descriptive name for this database.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">-dbtype</span></code></dt><dd><p>The type of input sequences — acceptable values are
<code class="docutils literal notranslate"><span class="pre">prot</span></code> (for protein) and <code class="docutils literal notranslate"><span class="pre">nucl</span></code> (for nucleotide)
sequences.</p>
</dd>
<dt><code class="docutils literal notranslate"><span class="pre">-out</span></code></dt><dd><p>The value here will be used to name the output files.
This is also the name you will need to use for the
database while doing a search
(see <a class="reference internal" href="#new-db-name"><span class="std std-ref">New DB Name</span></a>) in output.</p>
</dd>
</dl>
<p>When formatting is complete, you will notice the following
files in the <code class="docutils literal notranslate"><span class="pre">databases</span></code> directory:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ls -lh
</pre></div>
</div>
<p>Output:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="go">total 585M</span>
<span class="go">-rw-rw-r-- 1 user user 100M Mar 24 15:13 swissprot.phr</span>
<span class="go">-rw-rw-r-- 1 user user 4.4M Mar 24 15:13 swissprot.pin</span>
<span class="go">-rw-rw-r-- 1 user user 2.2M Mar 24 15:13 swissprot.pog</span>
<span class="go">-rw-rw-r-- 1 user user  18M Mar 24 15:13 swissprot.psd</span>
<span class="go">-rw-rw-r-- 1 user user 411K Mar 24 15:13 swissprot.psi</span>
<span class="go">-rw-rw-r-- 1 user user 195M Mar 24 15:13 swissprot.psq</span>
<span class="go">-rw-rw-r-- 1 user user 267M Feb 10 15:00 uniprot_sprot.fasta</span>
</pre></div>
</div>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="../../index.html">Linux for Biologists</a></h1>








<h3>Navigation</h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../about/index.html">1. About this book</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../using-linux/index.html">2. Getting started with Linux</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../getting-software/index.html">3. Getting software on Linux</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../index.html">4. Using the Linux command line</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../shell-terminal.html">4.1. Shell and Terminal</a></li>
<li class="toctree-l2"><a class="reference internal" href="../commands/index.html">4.2. Commands — an overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="../other-commands/index.html">4.3. Other useful commands</a></li>
<li class="toctree-l2"><a class="reference internal" href="../editing-text/index.html">4.4. Editing text files using nano</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="index.html">4.5. Exercise — using the command-line</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="summary.html">4.5.1. Summary of steps</a></li>
<li class="toctree-l3"><a class="reference internal" href="sample.html">4.5.2. Get sample data</a></li>
<li class="toctree-l3"><a class="reference internal" href="install.html">4.5.3. Install NCBI BLAST+ package</a></li>
<li class="toctree-l3"><a class="reference internal" href="query.html">4.5.4. Download query sequence</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">4.5.5. Download protein sequence database</a></li>
<li class="toctree-l3"><a class="reference internal" href="search.html">4.5.6. Search database using query sequence</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../notes.html">4.6. Notes</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../galaxy/index.html">5. Getting started with Galaxy</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../documentation/index.html">6. Documentation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../glossary.html">7. Glossary</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../credits.html">8. Credits</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../../index.html">Documentation overview</a><ul>
  <li><a href="../index.html"><span class="section-number">4. </span>Using the Linux command line</a><ul>
  <li><a href="index.html"><span class="section-number">4.5. </span>Exercise — using the command-line</a><ul>
      <li>Previous: <a href="query.html" title="previous chapter"><span class="section-number">4.5.4. </span>Download query sequence</a></li>
      <li>Next: <a href="search.html" title="next chapter"><span class="section-number">4.5.6. </span>Search database using query sequence</a></li>
  </ul></li>
  </ul></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2021, Vimalkumar Velayudhan.
      
      |
      <a href="../../_sources/command-line/exercise/database.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>